pyomo/contrib/iis/mis.py

```diff
@@ -0,0 +1,373 @@
+        for constr in elastic_filter:
+            constr.deactivate()
+            for var, val in _modified_model_value_cache.items():
+                var.set_value(val, skip_validation=True)
+            math_failure = False
+            try:
+                results = solver.solve(modified_model, tee=tee)
+            except:
+                math_failure = True
+
+            if math_failure:
+                constr.activate()
+                guards.append(constr)
+            elif pyo.check_optimal_termination(results):
+                constr.activate()
+                deletion_filter.append(constr)
+            else:  # still infeasible without this constraint
+                pass
+
+        msg += "Computed Minimal Intractable System (MIS)!\n"
+        msg += "Constraints / bounds in MIS:\n"
+        msg = _get_results(deletion_filter, msg)
+        msg += "Constraints / bounds in guards for stability:"
+        msg = _get_results(guards, msg)
+
+    logger.info(msg)
+
+
+def _get_results_with_value(constr_value_generator, msg=None):
+    # note that "lb_for_" and "ub_for_" are 7 characters long
+    if msg is None:
+        msg = ""
+    for c, value in constr_value_generator:
+        c_name = c.name
+        if "_variable_bounds" in c_name:
+            name = c.local_name
+            if "lb" in name:
+                msg += f"\tlb of var {name[7:]} by {value}\n"
+            elif "ub" in name:
+                msg += f"\tub of var {name[7:]} by {value}\n"
+            else:
+                raise RuntimeError("unrecognized var name")
+        else:
+            msg += f"\tconstraint: {c_name} by {value}\n"
+    return msg
+
+
+def _get_results(constr_generator, msg=None):
+    # note that "lb_for_" and "ub_for_" are 7 characters long
+    if msg is None:
+        msg = ""
+    for c in constr_generator:
+        c_name = c.name
+        if "_variable_bounds" in c_name:
+            name = c.local_name
+            if "lb" in name:
+                msg += f"\tlb of var {name[7:]}\n"
+            elif "ub" in name:
+                msg += f"\tub of var {name[7:]}\n"
+            else:
+                raise RuntimeError("unrecognized var name")
+        else:
+            msg += f"\tconstraint: {c_name}\n"
+    return msg
+
+
+def _get_constraint(modified_model, v):
+    if "_slack_plus_" in v.name:
+        constr = modified_model.find_component(v.local_name[len("_slack_plus_") :])
+        if constr is None:
+            raise RuntimeError(
+                "Bad constraint name {v.local_name[len('_slack_plus_'):]}"
+            )
+        return constr
+    elif "_slack_minus_" in v.name:
+        constr = modified_model.find_component(v.local_name[len("_slack_minus_") :])
+        if constr is None:
+            raise RuntimeError(
+                "Bad constraint name {v.local_name[len('_slack_minus_'):]}"
+            )
+        return constr
+    else:
+        raise RuntimeError("Bad var name {v.name}")
```

Comment:
Missing `f` for format strings in these error messages.

Reply:
Ooh! Those are bad. Thanks for catching them.